examples/graphene_apps/python_worker/graphene/graphene_sgx_attestation.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/python3
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+import logging
+import base64
+from avalon_worker.attestation.sgx_attestation import SGXAttestation
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+logger.addHandler(logging.StreamHandler(sys.stdout))
+
+# -------------------------------------------------------------------------
+
+
+class GrapheneSGXAttestation(SGXAttestation):
+    """
+    SGX Attestation implementation for Graphene.
+    """
+
+# -------------------------------------------------------------------------
+
+    # Class variables
+
+    # Graphene SGX TARGET INFO
+    GRAPHENE_SGX_TARGET_INFO_FILE = "/dev/attestation/my_target_info"
+    # Graphene SGX user report data
+    GRAPHENE_SGX_USER_REPORT_DATA_FILE = "/dev/attestation/user_report_data"
+    # Graphene SGX Quote file
+    GRAPHENE_SGX_QUOTE_FILE = "/dev/attestation/quote"
+
+# -------------------------------------------------------------------------
+
+    def get_mrenclave(self):
+        """
+        Get mrenclave value of enclave.
+
+        Returns :
+            mrenclave value of enclave as hex string.
+            If Intel SGX environment is not present returns empty string.
+        """
+        try:
+            # Read target info
+            target_info_file = \
+                GrapheneSGXAttestation.GRAPHENE_SGX_TARGET_INFO_FILE
+            with open(target_info_file, 'rb') as fd:
+                data = fd.read()
+                mrencalve = data[:32].hex()
+        except Exception:
+            logger.warn("Graphene-SGX Environment not setup."
+                        "Return empty mrencalve string.")
+            mrencalve = ""
```

Comment:
Put this at line 53. I reckon due to the except block it will be populated for sure but still better init at start.

Reply:
ok